hr_expense_cancel/models/hr_expense.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class HrExpenseSheet(models.Model):
+    _inherit = "hr.expense.sheet"
+
+    @api.multi
+    def action_cancel(self):
+        for sheet in self:
+            account_move = sheet.account_move_id
+            sheet.account_move_id = False
+
+            # If the sheet is paid then remove payments
+            if sheet.state == 'done':
+                payments = self.env['account.payment'].search([
+                    ('expense_sheet_id', '=', sheet.id),
+                    ('state', '!=', 'cancelled'),
+                ])
+                if sheet.expense_line_ids[0].payment_mode == 'own_account':
+                    self._remove_move_reconcile(payments, account_move)
+                    self._cancel_payments(payments)
+                else:
+                    # In this case, during the cancellation the journal entry
+                    # will be deleted
+                    self._cancel_payments(payments)
+                    payments.unlink()
+
+            if account_move.exists():
```

Comment:
Why do you need this check?

Reply:
I need this check because it is possible that in the previous steps where I cancel the payment associated with the 'expense sheet', this 'journal entry' (account_move) is deleted. This happens specifically when the payment_mode of the expenses of the sheet is 'company_account' (Payment by = Company), because in that case, the payment can be removed after canceling and removing this payment the journal entry (account_move) is also deleted, that's why I have to check if it has not been deleted yet to remove it then.

Reply:
Then please always add a comment in the "conflicting" lines that you already know that are subject of debates for avoiding the question.

Reply:
Well, I've already put a comment, check to see what you think